src/api/c/deconvolution.cpp

```diff
@@ -0,0 +1,135 @@
+/*******************************************************
+ * Copyright (c) 2017, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <af/image.h>
+#include <err_common.hpp>
+#include <backend.hpp>
+#include <Array.hpp>
+#include <arith.hpp>
+#include <cast.hpp>
+#include <fftconvolve.hpp>
+#include <handle.hpp>
+#include <reduce.hpp>
+#include <transpose.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <type_traits>
+
+using af::dim4;
+using namespace detail;
+
+template<typename T>
+Array<T> vancittert(const Array<T>& I, const Array<T>& prev, const Array<T>& P)
+{
+    using cplxtype = typename std::conditional<std::is_same<T, double>::value, cdouble, cfloat>::type;
+
+    auto convFn = fftconvolve<T, T, cplxtype, std::is_same<T, double>::value, std::is_integral<T>::value, (dim_t)2>;
+
+    auto cnv = convFn(prev, P, false, AF_BATCH_NONE);
+
+    return arithOp<T, af_sub_t>(I, cnv, I.dims());
+}
+
+template<typename T>
+Array<T> jansson(const Array<T>& dist, const Array<T>& prev)
+{
+    T mx = reduce_all<af_max_t, T, T>(prev, true); //true indicates replace nan with 0
+    auto dims   = dist.dims();
+    auto mxArr  = createValueArray(dims, mx);
+    auto ONES   = createValueArray(dims, scalar<T>(1));
+    auto HALF   = createValueArray(dims, scalar<T>(0.5));
+    auto midVal = arithOp<T, af_mul_t>(mxArr, HALF, dims);      // B/2
+    auto diff   = arithOp<T, af_sub_t>(prev, midVal, dims);     // |I(n-1) - B/2|
```

Comment:
missed that. Need to look into why tests passed, or if the abs is needed or not. Thanks